src/param.c

```diff
@@ -142,6 +142,71 @@ stumpless_new_param( const char *name, const char *value ) {
   return result;
 }
 
+struct stumpless_param *
+stumpless_new_param_from_string( const char *string ) {
+
+  VALIDATE_ARG_NOT_NULL( string );
+
+  int i;
+  size_t name_len = 0;
+  size_t name_start = 0;
+  size_t value_len = 0;
+  size_t value_start = 0;
+
+  /* Check that the characters in 'name' are allowed. */
+  for (i = 0; string[i] != '='; i++) {
+    if (!((string[i] >= 'a' && string[i] <= 'z') ||
+	  (string[i] >= 'A' && string[i] <= 'Z') ||
+	  (string[i] >= '0' && string[i] <= '9') ||
+	  (string[i] == '-') ||
+	  (string[i] == '.') ||
+	  (string[i] == '_'))){
+      raise_invalid_param(  );
+      return NULL;
+    }
+    if (string[i] == '\0') {
+      raise_invalid_param(  );
+      return NULL;
+    }
+  }
+  
+  /* Check if the character after the '=' is '"', else raise an error. */
+  if (string[i + 1] != '"'){
+    raise_invalid_param(  );
+    return NULL;
+  }
+
+  name_len = i;
+  /* Declare an array to hold the 'name' string.
+     We add 1 to the length for the '\0' character. */
+  char name[name_len + 1];
```

Comment:
The Windows builds appear to be failing on these array declarations (this one and line 200). You'll likely need to change these arrays to be dynamically allocated on the heap (via `alloc_mem` and subsequent `free_mem`). I believe the dynamic size of the array based on a variable, while supported in some compilers like GCC, is not universal.

You could keep the name buffer on the stack if you size it to `STUMPLESS_MAX_PARAM_NAME_LENGTH+1` and do an additional length check when scanning the name portion of the string. The value can theoretically be any size and will therefore need to be dynamic. Windows does support dynamic stack-based buffers via `_malloca` and `_freea`, but using this in a portable manner is probably more work than it's worth for this change.

Reply:
I think its better I just allocate on the heap through ```alloc_mem```.